src/Symfony/Bridge/Twig/Extension/IntlExtension.php

```diff
@@ -0,0 +1,325 @@
+     * @param string|null $displayLocale
+     *
+     * @return null|string
+     */
+    public function getLocalizedCountry($country, $displayLocale = null)
+    {
+        return Intl::getRegionBundle()->getCountryName((string) $country, $displayLocale ?: $this->getDefaultLocale());
+    }
+
+    /**
+     * Returns a localized currency amount for given number and currency code.
+     *
+     * @param mixed       $amount
+     * @param string      $currency
+     * @param string|null $displayLocale
+     *
+     * @return null|string
+     */
+    public function getLocalizedCurrency($amount, $currency, $displayLocale = null)
+    {
+        $formatter = $this->createNumberFormatter(
+            $displayLocale ?: $this->getDefaultLocale(),
+            'currency'
+        );
+
+        return $formatter->formatCurrency((float) $amount, $currency);
+    }
+
+    /**
+     * Returns a localized currency amount for a given number and currency code.
+     *
+     * @param string      $currency
+     * @param string|null $displayLocale
+     *
+     * @return null|string
+     */
+    public function getLocalizedCurrencyName($currency, $displayLocale = null)
+    {
+        return Intl::getCurrencyBundle()->getCurrencyName((string) $currency, $displayLocale ?: $this->getDefaultLocale());
+    }
+
+    /**
+     * Returns a currency symbol for a given code.
+     *
+     * @param string      $currency
+     * @param string|null $displayLocale
+     *
+     * @return null|string
+     */
+    public function getCurrencySymbol($currency, $displayLocale = null)
+    {
+        return Intl::getCurrencyBundle()->getCurrencySymbol((string) $currency, $displayLocale ?: $this->getDefaultLocale());
+    }
+
+    /**
+     * Returns a date formatted for a given locale, and options.
+     *
+     * @param \DateTimeInterface|string $date
+     * @param string                    $dateFormat
+     * @param string                    $timeFormat
+     * @param string|null               $displayLocale
+     * @param \DateTimeZone|string|null $timezone
+     * @param string|null               $format
+     * @param string                    $calendar
+     *
+     * @return bool|string
+     *
+     * @throws \Exception|\InvalidArgumentException If the date or a format is invalid
+     */
+    public function getLocalizedDate($date, $dateFormat = 'medium', $timeFormat = 'medium', $displayLocale = null, $timezone = null, $format = null, $calendar = 'gregorian')
+    {
+        if (is_string($date)) {
+            $date = new \DateTime($date);
+        }
+
+        if (!$date instanceof \DateTimeInterface) {
+            throw new \InvalidArgumentException(sprintf('Expected a \DateTimeInterface or time string got "%s"', is_object($date) ? get_class($date) : gettype($date)));
+        }
+
+        if (false !== $timezone) {
+            if (!$timezone instanceof \DateTimeZone) {
+                $timezone = new \DateTimeZone($timezone ?: $this->getDefaultTimezone());
```

Comment:
you should be using the Twig function about converting dates to a DateTime in the expected timezone